packages/onboarding/src/mshots-image/index.tsx

```diff
@@ -35,30 +35,31 @@ export function mshotsUrl( targetUrl: string, options: MShotsOptions, count = 0
 
 const MAXTRIES = 10;
 
-// This custom react hook returns undefined while the image is loading and
-// a HTMLImageElement (i.e. the class you get from `new Image()`) once loading
-// is complete.
+// This custom react hook returns null while the image is loading and the page
+// (not image) URL once loading is complete.
 //
 // It also triggers a re-render (via setState()) when the value changes, so just
-// check if it's truthy and then treat it like any other Image.
+// check that the requested URL matches the returned URL.
 //
-// Note the loading may occur immediately and synchronously if the image is
-// already or may take up to several seconds if mshots has to generate and cache
-// new images.
+// Note the loading may occur immediately if the image is already available, or
+// may take several seconds if mshots has to generate and cache new images.
 //
 // The calling code doesn't need to worry about the details except that you'll
 // want some sort of loading display.
 //
 // Inspired by https://stackoverflow.com/a/60458593
-const useMshotsImg = ( src: string, options: MShotsOptions ): HTMLImageElement | undefined => {
-	const [ loadedImg, setLoadedImg ] = useState< HTMLImageElement >();
+const useMshotsImg = (
+	src: string,
+	options: MShotsOptions,
+	imgRef: React.MutableRefObject< HTMLImageElement | null >
+): string | null => {
+	const [ loadedImg, setLoadedImg ] = useState< string | null >( null );
```

Comment:
Not important: this probably would be such a neat `useQuery` with `select` prop.

Reply:
I don't have a ton of experience with `useQuery`, so I'm probably going to skip this one, if that's okay 😶